pkg/sdk/agent.go

```diff
@@ -151,17 +154,28 @@ func (sdk *agentSDK) Attestation(ctx context.Context, reportData [size64]byte) (
 	return response.File, nil
 }
 
-func signData(userID string, privKey *rsa.PrivateKey) ([]byte, error) {
+func signData(userID string, privKey any) ([]byte, error) {
```

Comment:
rather than any private key use https://pkg.go.dev/crypto#Signer interface

Reply:
and since all private keys implement the signer method, no need to switch between them, you can simply call the sign method